src/lib.rs

```diff
@@ -454,6 +464,8 @@ pub enum Format {
     MatchVariant(Expr, Vec<(Pattern, String, Format)>),
     /// Format generated dynamically
     Dynamic(DynFormat),
+    /// Apply a dynamic format from a named variable in the scope
+    Apply(String),
```

Comment:
We could use a `Cow<'static, str>` here instead to optimize for cases where the label is a string literal we already know. This pattern can be applied more largely to other `String` signatures and fields in the codebase, but at minimum this is a fairly innocuous swap-in.

Reply:
I think I'd prefer to avoid doing that unless we have reason to believe that the string is being frequently copied.